File: skellig-test-step-processing-db/src/main/java/org/skellig/teststep/processor/db/model/DatabaseDetails.java
package org.skellig.teststep.processor.db.model;

import java.util.Optional;

public class DatabaseDetails {

    private String serverName;
    private String userName;
    private String password;

    public DatabaseDetails() {
    }

    public DatabaseDetails(String serverName, String userName, String password) {
        this.serverName = serverName;
        this.userName = userName;
        this.password = password;
    }

    public String getServerName() {
        return serverName;
    }

    public Optional<String> getUserName() {
        return Optional.ofNullable(userName);
    }

    public Optional<String> getPassword() {
        return Optional.ofNullable(password);
    }
}

File: skellig-test-step-processing-db/src/main/kotlin/org/skellig/teststep/processor/db/model/DatabaseDetails.kt
package org.skellig.teststep.processor.db.model

/**
 * Represents the details of a database server.
 * It contains the server name, username, and password.
 *
 * @param serverName The name of the database server.
 * @param userName The username to connect to the database server. Can be null.
 * @param password The password to connect to the database server. Can be null.
 */
open class DatabaseDetails(val serverName: String,
                           val userName: String?,
                           val password: String?) {
    override fun toString(): String {
        return "serverName = '$serverName', userName = $userName, password = $password"
    }
}